### DISCUSSION TITLE
UVtools birthday 🎁 2 Years 🥳

### DISCUSSION THREAD

**Discussion by sn4k3** [sn4k3/UVtools#451] [Announcements]
# Today (06/04) it's the UVtools birthday 🎁
## IT DOES 2 YEARS OLD 🥳

Just 2 years has passed from first release, short time, a new born, still it grow big and powerfull tool with an incredible performance!

## Some stats:

- Open-Source and Free
- Downloads: ± 63.2k
- Repo size: 547 MB
- Code size: 3.74 MB
- Code lines: ±145k
- Code contributors: 6
- Releases: 181
- Major releases: 3
- Commits: 916
- Traffic: ± 750 views and 250 unique views per day
- Issues: > 241 and solved
- Highest single donation: 50 USD (excluding fees)

## Some facts:

- Project born from a hobby wish to return to C# and try the new features of the language, but also because i had an ideia to buy a printer (EPAX) on the time, but not want to get restricted with slicers. I never did buy the printer... And i'm still an FDM guy.
- UVtools was former called PrusaSL1Viewer (http://github.com/sn4k3/PrusaSL1Viewer)
- First release was codded in a night and was only a simple file viewer for SL1 files from PrusaSlicer, it could only view slices and extract contents (unzip). It had only 2 files ([v0.1](https://github.com/sn4k3/UVtools/releases/tag/v0.1))
- Second release, 1 week after, incorporated cbdlp format, and allow file convertion from sl1 to cbdlp ([v0.2](https://github.com/sn4k3/UVtools/releases/tag/v0.2))
- After second release it start to grow exponential and get rich of features, CTB format was implemented on v0.3.1, 24 days after first release, give it a try to fell the difference ;)  ([v0.3.1](https://github.com/sn4k3/UVtools/releases/tag/v0.3.1))
- Many of the formats was only possible due uv3dp project, big thanks to Jason McMullan (https://github.com/ezrec/uv3dp)
- Just on [v0.5.1](https://github.com/sn4k3/UVtools/releases/tag/v0.5.1) project got renamed to UVtools
- Cancelling actions were not possible in the past
- Major version [1.0](https://github.com/sn4k3/UVtools/releases/tag/v1.0.0.0) got rewritten and add support for Linux and macOS
- Last year it was on [v2.8.2](https://github.com/sn4k3/UVtools/releases/tag/v2.8.2)
- Most of the features are requests from people, others are just my imagination...
- Just to show 1 layer, it need to execute more than 500 lines of code, considering that, is blazing fast, i'm always amazed...
- Issue detection is more than 1000 lines of code
- The developer (me) don't personally use UVtools, because i don't do resin printing, as so, he is a big noob on resin themes
- UVtools logo background is purple, guess why...
- UVtools is a huge toolkit but also provides a huge code base for adopters, samples and solutions to problems which in future people may use.
- Very permissive, so stupid that you can program C# inside UVtools, mess with your system and even format drives!
- UVtools may be the faster program that you can use to turn any photo in grayscale... Try to open any on it ;)
- It works like magic, i got beaten already by my own creation
- By now it took a lot of hours to reach this point, maintaining such project isn't easy, consider yourself a blessed person when using this software.

Please give a try on a old version to know it roots.
Have fun and make sure to use it today!

**Comment by X3msnake** [comment]
Awsome Master. keep up the great work :)
